tests/raop/test_raop_functional.py

```diff
@@ -135,6 +136,20 @@ async def test_stream_complete_legacy_auth(
     assert raop_state.auth_setup_performed == require_auth
     assert audio_matches(raop_state.raw_audio, frames=10)
 
+@pytest.mark.parametrize(
+    "raop_properties,raop_server_password,raop_client_password",
+    [({"et": "0"}, "test", "test"), ({"et": "0"}, None, None)],
+)
+async def test_stream_with_password(
```

Comment:
I miss some kind of verification that password validation was performed. Maybe you can add a bool that you set after a correct password has been provided, more or less like `auth_setup_performed`? It's not great, but it at least allows for some additional confidence that everything worked.

Reply:
I think the nicer way would be to catch the `AuthenticationError`, since that is the expected behaviour. Sadly putting a try catch around the call doesn't seem to work. Does the `async` stuff somehow prevent the correct propagation of exceptions ?

Reply:
Sure, you can do that as well. But there won't be any exception when the password is correct, so I guess you need separate test cases for positive contra negative outcome?

Shouldn't be any problem with async in regards to exceptions. There are a few layers in the cake we know as `stream_file`... Maybe it is swallowed somewhere along the way and not propagated back to the caller properly (by accident)?

Reply:
I don't know, what I made wrong the first time, but this time, I have no problem catching the exception this time... What do you think about the change ?

Reply:
Yep, that works! 👍